### ISSUE TITLE
add a `const` directive with a const value

### ISSUE THREAD

**Issue by tusharmath** [tailcallhq/tailcall#373]
**Description**
A `const` directive would basically always resolve with a constant value. The value will be provided within the directive. This is useful to writing tests and sometimes exposing constant values.

**Usage**
```graphql
schema {
  query: Query
}

type Query {
  user: User @const(data: {name: "John", age: 12})
}

type User {
  name: String
  age: Int
}
```

**Expected Behaviour**
- The provided `data` should always match the shape of the type provided. For example 
  
  **correct**
  ```graphql
  type Query {
    user: User @const(data: {name: "John", age: 12})
  }
  
  type User {
    name: String!
    age: Int!
  }
  ```
  
  **wrong**
  ```graphql
  type Query {
    user: User @const(data: {name: "John"})
  }
  
  type User {
    name: String!
    age: Int!
  }
  ```

**NOTE**
- Make sure existing functionality doesn't break
- Add tests in the `tests/` dir.

**Comment by tusharmath** [comment]
@akansh-bansal Updated the expected behaviour — 

> The output schema should mark the type of the field as required.

**Comment by amitksingh1490** [comment]
Fixed by #408
